src/server/v2.0/handler/security.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+
+	"github.com/go-openapi/runtime/middleware"
+
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	securityModel "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/securityhub"
+
+	"github.com/goharbor/harbor/src/common/security"
+	"github.com/goharbor/harbor/src/controller/securityhub"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/pkg/scan/dao/scan"
+	secHubModel "github.com/goharbor/harbor/src/pkg/securityhub/model"
+)
+
+func newSecurityAPI() *securityAPI {
+	return &securityAPI{
+		controller: securityhub.Ctl,
+	}
+}
+
+type securityAPI struct {
+	BaseAPI
+	controller securityhub.Controller
+}
+
+func (s *securityAPI) GetSecuritySummary(ctx context.Context,
+	params securityModel.GetSecuritySummaryParams) middleware.Responder {
+	secCtx, ok := security.FromContext(ctx)
+	if !ok {
+		return s.SendError(ctx, errors.UnauthorizedError(errors.New("security context not found")))
+	}
+	if !secCtx.IsAuthenticated() && !secCtx.IsSysAdmin() {
```

Comment:
I think the logic may have some potential issue if we only want open this API for system admin.